src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -482,6 +482,27 @@ private WriteConcern GetTransactionWriteConcern()
                 WriteConcern.WMajority;
         }
 
+        private WriteConcern GetCommitTransactionWriteConcern(bool isCommitARetry)
+        {
+            var writeConcern = GetTransactionWriteConcern();
+            if (isCommitARetry)
+            {
+                return writeConcern.WTimeout == null
+                    ? writeConcern.With(mode:"majority", wTimeout: TimeSpan.FromMilliseconds(10000))
+                    : writeConcern.With("majority");
+            }
+            else
+            {
+                return writeConcern;
+            }
+        }
```

Comment:
Proposed refactoring:

    private WriteConcern GetCommitTransactionWriteConcern(bool isCommitARetry)
    {
        var writeConcern = GetTransactionWriteConcern();
        if (isCommitARetry)
        {
            writeConcern = writeConcern.With(mode: "majority");
            if (writeConcern.WTimeout == null)
            {
                writeConcern = writeConcern.With(wTimeout: TimeSpan.FromMilliseconds(10000));
            }
        }

        return writeConcern;
    }

This reads more like the English explanation of what's going, namely:

* Start with the usual write concern
    * If commit is a retry then
        * replace the mode with "majority"
        * in addition if no wTimeout is specified set wTimeout to 10000 milliseconds

Reply:
Okay, taking your suggestion!